api/v1/storageprofile_types.go

```diff
@@ -92,3 +97,13 @@ const (
 func init() {
 	SchemeBuilder.Register(&StorageProfile{}, &StorageProfileList{})
 }
+
+func (sp *StorageProfile) GetSpecHash() string {
```

Comment:
Hash is also a verb, so maybe `hashSpec`? (just a suggestion)

Reply:
Ok, so if I go with that mindset I would say you are not getting anything, you are calculating. hanch: 
`calcSpecHash` fits better.